src/Socket.cpp

```diff
@@ -13,9 +14,9 @@ Socket::Socket(const int fd)
 {
```

Comment:
change name fd to server_fd and it shouldn't be const

Reply:
also the inet_ntop shouldn't be done in the constructor

Reply:
made a seperate funciton that sets up the client socket, which is called in the constructor of the client class that it only happens after the whole socket is already created.

Reply:
great work